src/vs/workbench/contrib/chat/browser/chatDragAndDrop.ts

```diff
@@ -122,12 +125,19 @@ export class ChatDragAndDrop extends Themable {
 		}
 	}
 
-	private getAttachContext(e: DragEvent): IChatRequestVariableEntry[] {
+	private async getAttachContext(e: DragEvent): Promise<IChatRequestVariableEntry[]> {
 		switch (this.getActiveDropType(e)) {
 
 			case ChatDragAndDropType.FILE: {
 				const data = extractEditorsDropData(e);
-				const contexts = data.map(editorInput => getEditorAttachContext(editorInput));
+				const contexts = await Promise.all(data.map(async editorInput => {
+					if (editorInput.resource && /\.(png|jpg|jpeg)$/i.test(editorInput.resource.path)) {
+						const fileBuffer = await this.fileService.readFile(editorInput.resource);
+						return getImageAttachContext(editorInput.resource, fileBuffer.value);
+					}
```

Comment:
If we keep images part of the file drop type, I would move this if/else logic into `getEditorAttachContext`. I didn't pick a great name for the function, maybe we need to change it to something like `getContextFromInput`